bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,881 @@
+		caseSensitiveSearchButton = new ToolItem(searchTools, SWT.CHECK);
+		caseSensitiveSearchButton.setImage(FindReplaceOverlayImages.get(FindReplaceOverlayImages.OBJ_CASE_SENSITIVE));
+		caseSensitiveSearchButton.setToolTipText(FindReplaceMessages.FindReplaceOverlay_caseSensitiveButton_toolTip);
+		caseSensitiveSearchButton.setSelection(findReplaceLogic.isActive(SearchOptions.CASE_SENSITIVE));
+		caseSensitiveSearchButton.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			activateInFindReplacerIf(SearchOptions.CASE_SENSITIVE, caseSensitiveSearchButton.getSelection());
+			updateIncrementalSearch();
+		}));
+	}
+
+	private void createWholeWordsButton() {
+		wholeWordSearchButton = new ToolItem(searchTools, SWT.CHECK);
+		wholeWordSearchButton.setImage(FindReplaceOverlayImages.get(FindReplaceOverlayImages.OBJ_WHOLE_WORD));
+		wholeWordSearchButton.setToolTipText(FindReplaceMessages.FindReplaceOverlay_wholeWordsButton_toolTip);
+		wholeWordSearchButton.setSelection(findReplaceLogic.isActive(SearchOptions.WHOLE_WORD));
+		wholeWordSearchButton.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			activateInFindReplacerIf(SearchOptions.WHOLE_WORD, wholeWordSearchButton.getSelection());
+			updateIncrementalSearch();
+		}));
+	}
+
+	private void createReplaceTools() {
+		Color warningColor = JFaceColors.getErrorText(getShell().getDisplay());
+
+		replaceTools = new ToolBar(replaceContainer, SWT.HORIZONTAL);
+		GridDataFactory.fillDefaults().grab(false, true).align(GridData.CENTER, GridData.END).applyTo(replaceTools);
+		replaceButton = new ToolItem(replaceTools, SWT.PUSH);
+		replaceButton.setImage(FindReplaceOverlayImages.get(FindReplaceOverlayImages.OBJ_REPLACE));
+		replaceButton.setToolTipText(FindReplaceMessages.FindReplaceOverlay_replaceButton_toolTip);
+		replaceButton.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			if (getFindString().isEmpty()) {
+				showUserFeedback(warningColor, true);
+				return;
+			}
+			performSingleReplace();
+			evaluateFindReplaceStatus();
+		}));
+		replaceAllButton = new ToolItem(replaceTools, SWT.PUSH);
+		replaceAllButton.setImage(FindReplaceOverlayImages.get(FindReplaceOverlayImages.OBJ_REPLACE_ALL));
+		replaceAllButton.setToolTipText(FindReplaceMessages.FindReplaceOverlay_replaceAllButton_toolTip);
+		replaceAllButton.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			if (getFindString().isEmpty()) {
+				showUserFeedback(warningColor, true);
+				return;
+			}
+			performReplaceAll();
+			evaluateFindReplaceStatus();
+		}));
+	}
+
+	private void createSearchBar() {
+		searchBar = new Text(searchBarContainer, SWT.SINGLE);
+		GridDataFactory.fillDefaults().grab(true, false).align(GridData.FILL, GridData.END).applyTo(searchBar);
+		searchBar.forceFocus();
+		searchBar.selectAll();
+		searchBar.addModifyListener(new ModifyListener() {
+			@Override
+			public void modifyText(ModifyEvent e) {
```

Comment:
Can this be a lambda?

Reply:
No, `ModifyListener` does not provide a factory to make this a Lambda.
![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/ee7a1eb2-0a15-4f8a-a92f-7b1635a2b950)
![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/bfc9eac9-8277-4e34-b72a-3e6e8644de65)

Reply:
What @HannesWell means is that `ModifyListener` is a functional interface, so you can simply pass a lambda that matches that interface. In this case, it would be something like:
```java
searchBar.addModifyListener(e -> {
  ...
});

Reply:
> No, `ModifyListener` does not provide a factory to make this a Lambda.

That's right, but the interface has only one method. Therefore it can be created as a lambda directly, without the need for a factory:
```
searchBar.addModifyListener(e -> {
  ...
});
```

Reply:
As discussed down below, there was still a way to simplify the listener. Fixed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/0a50625bd1074be023a70b085124f92ddcaa83f5